### ISSUE TITLE
Burp on freebsd: Certificate doesn't verify vs problem reading in extra_comms

### ISSUE THREAD

**Issue by do11** [grke/burp#657]
I installed burp-2.0.54 on freebsd and burp-2.0.54 on linux. Other linux clients work good with linux server (which is on Centos 6). But when connected from FreeBSD 8.0-STABLE I get such errors:

```
0bunny/etc/burp(340)# burp -al
2017-12-08 21:03:08: burp[48491] Could not find ssl_cert /etc/burp/ssl_cert-client.pem: No such file or directory
2017-12-08 21:03:08: burp[48491] Could not find ssl_key /etc/burp/ssl_cert-client.key: No such file or directory
2017-12-08 21:03:08: burp[48491] Could not find ssl_cert_ca /etc/burp/ssl_cert_ca.pem: No such file or directory
2017-12-08 21:03:08: burp[48491] auth ok
2017-12-08 21:03:08: burp[48491] Server version: 2.0.54
2017-12-08 21:03:08: burp[48491] Server will sign a certificate request
2017-12-08 21:03:08: burp[48491] Generating SSL key and certificate signing request
2017-12-08 21:03:08: burp[48491] Running '/usr/sbin/burp_ca --key --keypath /etc/burp/ssl_cert-client.key --request --requestpath /etc/burp/CA-client/bunny.x.net.csr --name bunny.x.net'
generating key bunny.x.net: /etc/burp/ssl_cert-client.key
Generating RSA private key, 2048 bit long modulus
............................+++
..............+++
e is 65537 (0x10001)
generating request bunny.x.net
2017-12-08 21:03:08: burp[48491] /usr/sbin/burp_ca returned: 0
2017-12-08 21:03:08: burp[48491] Sent /etc/burp/CA-client/bunny.x.net.csr
2017-12-08 21:03:09: burp[48491] Received: /etc/burp/ssl_cert-client.pem.48491
2017-12-08 21:03:09: burp[48491] Received: /etc/burp/ssl_cert_ca.pem.48491
2017-12-08 21:03:09: burp[48491] Rewriting conf file: /etc/burp/burp.conf
2017-12-08 21:03:09: burp[48491] Re-opening connection to server

2017-12-08 21:03:14: burp[48491] auth ok
2017-12-08 21:03:14: burp[48491] Server version: 2.0.54
2017-12-08 21:03:14: burp[48491] nocsr ok
2017-12-08 21:03:14: burp[48491] SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1

2017-12-08 21:03:14: burp[48491] Certificate doesn't verify.
2017-12-08 21:03:14: burp[48491] check cert failed
1bunny/etc/burp(341)#
1bunny/etc/burp(341)# burp -al
2017-12-08 21:03:17: burp[48529] auth ok
2017-12-08 21:03:17: burp[48529] Server version: 2.0.54
2017-12-08 21:03:17: burp[48529] nocsr ok
2017-12-08 21:03:17: burp[48529] SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1

2017-12-08 21:03:17: burp[48529] Certificate doesn't verify.
2017-12-08 21:03:17: burp[48529] check cert failed
```

At the same time on server:
```
Dec  8 20:48:17 fork burp[511634]: Connect from peer: ::ffff:192.1.1.1:39433
Dec  8 20:48:17 fork burp[511634]: forked child: 403863
Dec  8 20:48:17 fork burp[403863]: auth ok for: bunny.x.net
Dec  8 20:48:17 fork burp[403863]: Client bunny.x.net wants a certificate signed
Dec  8 20:48:17 fork burp[403863]: Received: /etc/burp/CA/bunny.x.net.csr
Dec  8 20:48:17 fork burp[403863]: Signing certificate signing request from bunny.x.net
Dec  8 20:48:17 fork burp[403863]: Running '/usr/sbin/burp_ca --name bunny.x.net --ca burpCA --sign --batch --dir /etc/burp/CA --config /etc/burp/CA.cnf'
Dec  8 20:48:17 fork burp[403863]: /usr/sbin/burp_ca returned: 0
Dec  8 20:48:17 fork burp[403863]: Sent /etc/burp/CA/bunny.x.net.crt
Dec  8 20:48:17 fork burp[403863]: Sent /etc/burp/ssl_cert_ca.pem
Dec  8 20:48:17 fork burp[403863]: Signed and returned client certificate request for bunny.x.net
Dec  8 20:48:17 fork burp[403863]: exit child
Dec  8 20:48:17 fork burp[511634]: pipe from child: end of data
Dec  8 20:48:17 fork burp[511634]: pipe from child: disconnected fd 7

Dec  8 20:48:23 fork burp[511634]: Connect from peer: ::ffff:192.1.1.1:36152
Dec  8 20:48:23 fork burp[511634]: forked child: 403941
Dec  8 20:48:23 fork burp[403941]: auth ok for: bunny.x.net
Dec  8 20:48:23 fork burp[403941]: Client bunny.x.net does not want a certificate signed
Dec  8 20:48:23 fork burp[403941]: SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1#012
Dec  8 20:48:23 fork burp[403941]: main socket: Peer closed SSL session
Dec  8 20:48:23 fork burp[403941]: problem reading in extra_comms
Dec  8 20:48:23 fork burp[403941]: running extra comms failed on server
Dec  8 20:48:23 fork burp[403941]: main socket: SSL write problem in asfd_do_write_ssl: 1 - 0=Success
Dec  8 20:48:23 fork error:140D00CF:SSL routines:SSL_write:protocol is shutdown:ssl_lib.c:999:

Message from syslogd@fork at Dec  8 20:48:23 ...
 error:140D00CF:SSL routines:SSL_write:protocol is shutdown:ssl_lib.c:999:
Dec  8 20:48:23 fork burp[403941]: main socket: Peer closed SSL session
Dec  8 20:48:23 fork burp[403941]: exit child
Dec  8 20:48:23 fork burp[511634]: pipe from child: end of data
Dec  8 20:48:23 fork burp[511634]: pipe from child: disconnected fd 7

Dec  8 20:48:26 fork burp[511634]: Connect from peer: ::ffff:192.1.1.1:19601
Dec  8 20:48:26 fork burp[511634]: forked child: 403974
Dec  8 20:48:26 fork burp[403974]: auth ok for: bunny.x.net
Dec  8 20:48:26 fork burp[403974]: Client bunny.x.net does not want a certificate signed
Dec  8 20:48:26 fork burp[403974]: SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1#012
Dec  8 20:48:26 fork burp[403974]: main socket: Peer closed SSL session
Dec  8 20:48:26 fork burp[403974]: problem reading in extra_comms
Dec  8 20:48:26 fork burp[403974]: running extra comms failed on server
Dec  8 20:48:26 fork burp[403974]: main socket: SSL write problem in asfd_do_write_ssl: 1 - 0=Success
Dec  8 20:48:26 fork error:140D00CF:SSL routines:SSL_write:protocol is shutdown:ssl_lib.c:999:
Message from syslogd@fork at Dec  8 20:48:26 ...
 error:140D00CF:SSL routines:SSL_write:protocol is shutdown:ssl_lib.c:999:
Dec  8 20:48:26 fork burp[403974]: main socket: Peer closed SSL session
Dec  8 20:48:26 fork burp[403974]: exit child
Dec  8 20:48:26 fork burp[511634]: pipe from child: end of data
Dec  8 20:48:26 fork burp[511634]: pipe from child: disconnected fd 7
```

ps. I censored only real hostname and IP.

**Comment by grke** [comment]
Hello,
Are you able to try recent master?

**Comment by do11** [comment]
No, I did not try, since server is using 2.0.59 and upgrading client would require upgrade server, and same operation on 1.4 branch required upgrading all clients again and restart backups, since it was somehow incompatible.

Aren't 2.0 is supported branch?

**Comment by grke** [comment]
Hello,

I haven't made a 2.0.59 release.

2.0.54 is the 'stable' release. That means I don't keep applying changes to it so that you have to keep upgrading. I am nearly ready to make a 2.1.x 'stable' release.

It is possible that I already fixed the problem that you are having in master.
So, I would appreciate it if you could confirm whether or not you can reproduce it in master.

I asked 'are you able' because for me to do it, I have to set up your exact environment and try to replicate the issue, which is very time consuming because I don't have the operating systems that you are using, and I am not necessarily experienced in their configuration. Especially freebsd.

I imagine it would be easier for you to set them up and try it out on some kind of VM, since you presumably have already got copies those operating systems and have some experience of their package managers and how to set them up. Hence 'are you able to try'?

In my experience, the most likely outcome is either that I take an hour or so to set up these machines, and find that either I can't replicate it at all, or that I can replicate it on 2.0.54, and it is fixed in master.

**Comment by do11** [comment]
Excuse me for typo, server is using 2.0.54.

**Comment by grke** [comment]
I think this is most likely OK in master, so I am closing until somebody finds out anything different.

**Comment by do11** [comment]
Is 2.1.24 will be enough to test? git master does not have `configure` built and I am not going to recreate appropriate autoconf on that legacy box. I tried to extract `git clone` over untarred 2.1.24 and use that configure scripts. Below is the result.

Also, upgraded (linux) server to burp 2.1.24 release from 2.0.54 release, hope it's safe.

Run of `burp -al` on freebsd client:
```
0bunny~/burp-github/burp-master(482)# ./burp -al
2017-12-25 17:48:05 +0300: burp[79086] Connecting to fork.x.ru:4971
2017-12-25 17:48:05 +0300: burp[79086] auth ok
2017-12-25 17:48:05 +0300: burp[79086] Server version: 2.1.24
2017-12-25 17:48:05 +0300: burp[79086] nocsr ok
2017-12-25 17:48:05 +0300: burp[79086] SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1

2017-12-25 17:48:05 +0300: burp[79086] Certificate doesn't verify.
2017-12-25 17:48:05 +0300: burp[79086] check cert failed
```

**Comment by do11** [comment]
Deleted client certs on both client and server:
```
0bunny~/burp-github/burp-master(491)# ./burp -al
2017-12-25 17:52:16 +0300: burp[79351] Could not find ssl_cert /etc/burp/ssl_cert-client.pem: No such file or directory
2017-12-25 17:52:16 +0300: burp[79351] Could not find ssl_key /etc/burp/ssl_cert-client.key: No such file or directory
2017-12-25 17:52:16 +0300: burp[79351] Could not find ssl_cert_ca /etc/burp/ssl_cert_ca.pem: No such file or directory
2017-12-25 17:52:16 +0300: burp[79351] Connecting to fork.x.ru:4971
2017-12-25 17:52:16 +0300: burp[79351] auth ok
2017-12-25 17:52:16 +0300: burp[79351] Server version: 2.1.24
2017-12-25 17:52:16 +0300: burp[79351] Server will sign a certificate request
2017-12-25 17:52:16 +0300: burp[79351] Generating SSL key and certificate signing request
2017-12-25 17:52:16 +0300: burp[79351] Running '/usr/sbin/burp_ca --key --keypath /etc/burp/ssl_cert-client.key --request --requestpath /etc/burp/CA-client/bunny.x.net.csr --name bunny.x.net'
generating key bunny.x.net: /etc/burp/ssl_cert-client.key
Generating RSA private key, 2048 bit long modulus
..................+++
........................................+++
e is 65537 (0x10001)
generating request bunny.x.net
2017-12-25 17:52:16 +0300: burp[79351] /usr/sbin/burp_ca returned: 0
2017-12-25 17:52:16 +0300: burp[79351] Sent /etc/burp/CA-client/bunny.x.net.csr
2017-12-25 17:52:17 +0300: burp[79351] Received: /etc/burp/ssl_cert-client.pem.79351
2017-12-25 17:52:17 +0300: burp[79351] Received: /etc/burp/ssl_cert_ca.pem.79351
2017-12-25 17:52:17 +0300: burp[79351] Re-opening connection to server
2017-12-25 17:52:22 +0300: burp[79351] Connecting to fork.x.ru:4971
2017-12-25 17:52:22 +0300: burp[79351] auth ok
2017-12-25 17:52:22 +0300: burp[79351] Server version: 2.1.24
2017-12-25 17:52:22 +0300: burp[79351] nocsr ok
2017-12-25 17:52:22 +0300: burp[79351] SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1

2017-12-25 17:52:22 +0300: burp[79351] Certificate doesn't verify.
2017-12-25 17:52:22 +0300: burp[79351] check cert failed
```

**Comment by do11** [comment]
Also. Syslog on the linux server with 2.1.24 release have many burp lines connected into blob:
```
Dec 25 17:45:01 fork burp[557509]: Server is using protocol=0 (auto)
Dec 25 17:45:01 fork burp[557509]: Client supports being sent json counters.
Dec 25 17:45:01 fork burp[557509]: Client has set protocol=1
Dec 25 17:45:01 fork d: end of data#0122017-12-25 17:41:29 +0300: burp[521429] pipe from child: disconnected fd 6#0122017-12-25 17:41:32 +0300: burp[521429] Connect from peer: ::ffff:x:33472#0122017-12-25 17:41:32 +0300: burp[521429] 0/9 child processes running on port 4971#0122017-12-25 17:41:32 +0300: burp[521429] Child 1 available#0122017-12-25 17:41:32 +0300: burp[521429] forked child on port 4971: 555494#0122017-12-25 17:41:34 +0300: burp[521429] pipe from child: end of data#0122017-12-25 17:41:34 +0300: burp[521429] pipe from child: disconnected fd 6#0122017-12-25 17:44:01 +0300: burp[521429] Connect from peer: ::ffff:x:57562#0122017-12-25 17:44:01 +0300: burp[521429] 0/9 child processes running on port 4971#0122017-12-25 17:44:01 +0300: burp[521429] Child 1 available#0122017-12-25 17:44:01 +0300: burp[521429] forked child on port 4971: 556945#0122017-12-25 17:44:01 +0300: burp[521429] Connect from peer: ::ffff:x:48425#0122017-12-25 17:44:01 +0300: burp[521429] 1/9 child processes running on port 4971#0122017-12-25 17:44:01 +0300: burp[521429] Child 2 available#0122017-12-25 17:44:01 +0300: burp[521429] forked child on port 4971: 556947#0122017-12-25 17:44:01 +0300: burp[521429] pipe from child: end of data#0122017-12-25 17:44:01 +0300: burp[521429] pipe from child: disconnected fd 6#0122017-12-25 17:44:01 +0300: burp[521429] pipe from child: end of data#0122017-12-25 17:44:01 +0300: burp[521429] pipe from child: disconnected fd 7#0122017-12-25 17:44:01 +0300: burp[521429] Connect from peer: ::ffff:x:40496#0122017-12-25 17:44:01 +0300: burp[521429] 0/9 child processes running on port 4971#0122017-12-25 17:44:01 +0300: burp[521429] Child 1 available#0122017-12-25 17:44:01 +0300: burp[521429] forked child on port 4971: 556965#0122017-12-25 17:44:02 +0300: burp[521429] pipe from child: end of data#0122017-12-25 17:44:02 +0300: burp[521429] pipe from child: disconnected fd 6#0122017-12-25 17:45:01 +0300: burp[521429] Connect from peer: ::1:51510#0122017-12-25 17:45:01 +0300: burp[
Dec 25 17:45:01 fork burp[557509]: Running timer script: live.x.ru /tank/burp/live.x.ru/current /tank/burp reserved1 reserved2 20h Mon,Tue,Wed,Thu,Fri,00,01,02,03,04,05,19,20,21,22,23 Sat,Sun,00,01,02,03,04,05,06,07,08,17,18,19,20,21,22,23
Dec 25 17:45:01 fork burp[557509]: Out of timeband: Mon,Tue,Wed,Thu,Fri,00,01,02,03,04,05,19,20,21,22,23
Dec 25 17:45:01 fork burp[557509]: Out of timeband: Sat,Sun,00,01,02,03,04,05,06,07,08,17,18,19,20,21,22,23
Dec 25 17:45:01 fork burp[557509]: Last backup: 2017-12-24 17:00:01
Dec 25 17:45:01 fork burp[557509]: Next after : 2017-12-25 13:00:01 (interval 20h)
Dec 25 17:45:01 fork burp[557509]: /usr/share/burp/scripts/timer_script returned: 1
Dec 25 17:45:01 fork burp[557509]: Not running backup of live.x.ru
Dec 25 17:45:01 fork burp[557509]: exit child
Dec 25 17:45:01 fork burp[521429]: pipe from child: end of data
Dec 25 17:45:01 fork burp[521429]: pipe from child: disconnected fd 7
```

**Comment by do11** [comment]
@grke Is this not worth reopening? I tested all you asked with git version and problem still occurs.

**Comment by grke** [comment]
Yes, reopening. Thank you.

**Comment by grke** [comment]
Hello,

I have:
Centos 6.9 (burp server)
FreeBSD 9.3 (burp client)

FreeBSD is not 8.0, because I haven't downloaded that version yet. Maybe I don't need to.

I was having "Certificate doesn't verify" on the client side.
I realised that the freebsd openssl version was very old, so I updated that to 1.1.0c with 'pkg install openssl-devel' and recompiled burp. Maybe this wasn't part of the problem, because I continued to have "Certificate doesn't verify" on the client side after cleaning out everything and starting again.

So, I changed this section of code in burp src/ssl.c on the client side to debug it:
```
int x;
        if((x=SSL_get_verify_result(ssl))!=X509_V_OK)
        {
printf("x: %d\n", x);
                logp_ssl_err("Certificate doesn't verify.\n");
                return -1;
        }
```

After I recompiled and ran it, it printed "x: 9".
In /usr/include/openssl/x509_vfy.h, it has:
#define         X509_V_ERR_CERT_NOT_YET_VALID                   9

So, I looked at the clocks on both machines.
```
[root@centos ~]# date
Sun Dec 31 11:10:41 AEST 2017
```
```
[root@freebsd ~/burp]# date
Sun Dec 31 01:01:04 EST 2017
```
I realised that freebsd is complaining that the certificate start date is in the future.
So, I set the freebsd clock to be in the future:
```
[root@freebsd ~/burp]# date 1712311500
Sun Dec 31 15:00:00 EST 2017
```

After I did that, 'burp -a l' works fine. I could have also waited about 10 hours, and freebsd would have caught up.

I can run this on the server side to see what the certificate dates are:
```
[root@centos ~]# openssl x509 -in /etc/burp/ssl_cert-server.pem -noout -startdate
notBefore=Dec 31 00:37:57 2017 GMT
```


So, maybe you can check your clocks and certificate dates?

If this is not the problem, please add the debug to the client side and find out what the SSL_get_verify_result code is. You might even want to do that before checking the clocks.

**Comment by grke** [comment]
I will add the SSL_get_verify_result to the code base in master, as it would be super useful.

**Comment by do11** [comment]
Before report I googled and found that most frequent cause of this error was time difference, so I checked time and waited few days just in case. Today:

Client:
```
0bunny/etc/burp(505)# TZ=GMT date
Mon Jan  1 16:11:06 GMT 2018
0bunny/etc/burp(506)# openssl x509 -noout -startdate -in ssl_cert_ca.pem
notBefore=Aug 29 12:01:49 2017 GMT
0bunny/etc/burp(507)# openssl x509 -noout -startdate -in ssl_cert-client.pem
notBefore=Dec 25 14:36:12 2017 GMT
```

Server:
```
root@fork.x.ru:/etc/burp# TZ=GMS date
Mon Jan  1 15:54:52 GMS 2018
root@fork.x.ru:/etc/burp# openssl x509 -in /etc/burp/ssl_cert-server.pem -noout -startdate
notBefore=Aug 29 12:01:50 2017 GMT
root@fork.x.ru:/etc/burp# openssl x509 -in /etc/burp/CA/bunny.x.net.crt -noout -startdate
notBefore=Dec 25 14:36:12 2017 GMT
```
After that on client:
```
0bunny~/burp-github/burp-master(527)# ./burp -al
2018-01-01 19:17:27 +0300: burp[96760] Connecting to fork.relcomhost.ru:4971
2018-01-01 19:17:28 +0300: burp[96760] auth ok
2018-01-01 19:17:28 +0300: burp[96760] Server version: 2.1.24
2018-01-01 19:17:28 +0300: burp[96760] nocsr ok
2018-01-01 19:17:28 +0300: burp[96760] SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1

2018-01-01 19:17:28 +0300: burp[96760] Certificate doesn't verify (7).
2018-01-01 19:17:28 +0300: burp[96760] check cert failed
```
On that box 7 is defined as `# define         X509_V_ERR_CERT_SIGNATURE_FAILURE               7`

**Comment by grke** [comment]
OK, thank you.
When I have time, I will try to install freebsd 8.

**Comment by do11** [comment]
I think this could be `Signature Algorithm: sha256WithRSAEncryption` which is not supported on older freebsd 8's openssl, while cert is generated on centos 6 with newer openssl with sha2.

Is there way to generate sha1 signature for compatibility with old systems? (It's private intranet anyway.)

Self-signed `burpCA` cert is using `Signature Algorithm: sha1WithRSAEncryption`. Would be good to allow older boxes to use sha1.

**Comment by do11** [comment]
No, my hypotheses is not worked.

I changed `/etc/burp/CA.cnf` from `default_md              = sha256` to `default_md              = sha1` then `rm`'ed certs on the client and reconnected. After that `ssl_cert-client.pem` have `Signature Algorithm: sha1WithRSAEncryption`, but connection problem still remained, and with same ssl error code.

```
0bunny/etc/burp(561)# burp -ql
2018-01-03 04:27:41 +0300: burp[12293] Connecting to fork.x.ru:4971
2018-01-03 04:27:41 +0300: burp[12293] auth ok
2018-01-03 04:27:41 +0300: burp[12293] Server version: 2.1.24
2018-01-03 04:27:41 +0300: burp[12293] nocsr ok
2018-01-03 04:27:41 +0300: burp[12293] SSL is using cipher: DHE-RSA-AES256-SHA      SSLv3 Kx=DH       Au=RSA  Enc=AES(256)  Mac=SHA1

2018-01-03 04:27:41 +0300: burp[12293] Certificate doesn't verify (7).
2018-01-03 04:27:41 +0300: burp[12293] check cert failed

1bunny/etc/burp(562)# openssl x509 -noout -text -in ssl_cert-client.pem|grep Algo
        Signature Algorithm: sha1WithRSAEncryption
            Public Key Algorithm: rsaEncryption
    Signature Algorithm: sha1WithRSAEncryption

```

**Comment by grke** [comment]
OK.
It seems to me that '7' means something like that though. I think you have the right sort of idea.

**Comment by do11** [comment]
On that FreeBSD 8.0-STABLE default openssl is `0.9.8k 25 Mar 2009`. I installed openssl from ports, which is `1.0.0d 8 Feb 2011`. After that burp release 2.1.24 is able to connect to burp-2.1.24 server after cert negotiation. That's unfortunate since older servers even more need backups.

Only downside is burp-2.1.24 server on linux is still log to syslog without proper line termination making its messages single long line blob.

**Comment by grke** [comment]
Hello,

I'm glad you figured out that it was openssl versions.
I don't think there is anything that I can do about that.

But, I can probably do something about newlines.
Do you have an example of the syslog messages that you are talking about?

**Comment by do11** [comment]
They are there https://github.com/grke/burp/issues/657#issuecomment-353873927

**Comment by grke** [comment]
I think it is because the server forks child process but does not closelog/openlog for syslog straight away, so all the processes write to the same file descriptor. This messes up the syslog, as you are seeing.

I see the same thing in the syslog on my server.
I have tried a patch, and will let it run for a day to see if the problem goes away.

**Comment by grke** [comment]
I still get the concatenated output in my syslog, but I also realised that I have stdout=1 in my burp-server.conf.
I suspect that syslog is logging the stdout of daemon processes, so I have now set stdout=0 and will see if this solved the problem.

**Comment by grke** [comment]
Yes, setting stdout=0 solves the syslog problem.

**Comment by do11** [comment]
Thanks.

ps. I believe `stdout` option is wrongly placed in the config, which is only partly fixed in 2.1 branch with `-Q` option. Naive packagers forget (as in never know) to put `-Q` in init scripts when daemonizing burp server. It's inverted logic. Now if I want to run burp verbose on console (only case when it needs to be verbose, and where some `--verbose` option would be enough) I should each time update config to have `stdout=1` then not forget to turn it back.

Please add _option_ to turn on verbosity, so we could always keep `stdout=0` in config.

**Comment by grke** [comment]
OK, added separate request:
https://github.com/grke/burp/issues/675
